js/base-engine.js

```diff
@@ -203,6 +204,75 @@ function d20plusEngine () {
 		});
 	};
 
+	d20plus.engine.enhanceMacros = (openedMacroId) => {
+		const $dialog = $(`.dialog[data-macroid=${openedMacroId}]`);
+		if (!openedMacroId || !$dialog[0]) return;
+		const $macro = $dialog.find(`.macro.tokenizer`);
+		const $b20macro = $dialog.find(`.tokenizer.b20`);
+		const $name = $dialog.find("input.name");
+		const $checkbox = $dialog.find(".isjs")
+			.on("change", () => {
+				if ($checkbox.prop("checked")) $macro.parent().addClass("jsdialog");
+				else $macro.parent().removeClass("jsdialog");
+			});
+		const macro = currentPlayer.macros._byId[openedMacroId];
+		const script = d20plus.engine.decodeScript($macro.val());
+		if (script) {
+			$macro.parent().addClass("jsdialog");
+			$b20macro.val(script);
+			$checkbox.prop("checked", true);
+		} else {
+			$b20macro.val($macro.val());
+		}
+		$dialog.find(".btn.testmacro").on("click", () => {
+			if (!$checkbox.prop("checked")) {
+				$macro.val($b20macro.val());
+			} else {
+				$macro.val(d20plus.engine.runScript($b20macro.val(), macro));
+			}
+		});
+		const $buttons = $dialog.parent()
+			.find(".ui-dialog-buttonpane button:not(.active)")
+			.addClass("active");
+		$buttons.on("mouseup", () => {
+			let name = $name.val() || "Untitled";
+			const existing = d20.Campaign.players.map(p => p.macros
+				.filter(m => m.id !== openedMacroId && (p.id === d20_player_id || m.visibleToCurrentPlayer()))
+				.map(m => m.get("name"))).flat();
+			while (existing.includes(name)) name = name.replace(/(\d*?)$/, id => 1 * id + 1);
```

Comment:
`existing.includes(name)` -> `existing.has(name)` (to match `Set` change, above)

`1 * id + 1` -> `Number(id) + 1`

Reply:
done